src/inc/til/small_vector.h

```diff
@@ -357,57 +359,39 @@ namespace til
         small_vector(const small_vector& other) :
             small_vector{}
         {
-            operator=(other);
+            _copy_assign(other);
         }
 
         // NOTE: If an exception is thrown while copying, the vector is left empty.
         small_vector& operator=(const small_vector& other)
         {
-            clear();
-            reserve(other._size);
-
-            std::uninitialized_copy(other.begin(), other.end(), _uninitialized_begin());
-            _size = other._size;
+            if (this != &other)
+            {
+                clear();
+                _copy_assign(other);
+            }
 
             return *this;
         }
 
-        small_vector(small_vector&& other) noexcept :
-            small_vector{}
+        small_vector(small_vector&& other) noexcept
         {
-            operator=(std::move(other));
+            _move_assign(other);
         }
 
         small_vector& operator=(small_vector&& other) noexcept
         {
-            std::destroy(begin(), end());
-            if (_capacity != N)
+            if (this != &other)
             {
-                _deallocate(_data);
-            }
+                std::destroy(begin(), end());
+                if (_capacity != N)
+                {
+                    _deallocate(_data);
+                }
 
-            if (other._capacity == N)
-            {
-                _data = &_buffer[0];
-                _capacity = N;
-                _size = other._size;
-                // The earlier static_assert(std::is_nothrow_move_constructible_v<T>)
-                // ensures that we don't exit in a weird state with invalid `_size`.
-#pragma warning(suppress : 26447) // The function is declared 'noexcept' but calls function '...' which may throw exceptions (f.6).
-                std::uninitialized_move(other.begin(), other.end(), _uninitialized_begin());
-                std::destroy(other.begin(), other.end());
-            }
-            else
-            {
-                _data = other._data;
-                _capacity = other._capacity;
-                _size = other._size;
+                _move_assign(other);
             }
 
-            other._data = &other._buffer[0];
-            other._capacity = N;
-            other._size = 0;

```

Comment:
These 21 lines are moved 1:1 into `_move_assign`, for the same reason as for `_copy_assign`.